.github/workflows/check_new_kernel_configs.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+
+def get_data_from_config(input_file):
+    with open(input_file, 'r') as file:
+        input_config_data = file.read()
+    return input_config_data
+
+def check_kernel(input_file):
+    match = re.search(r'SPECS/(.*?)/', input_file)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+def check_config_arch(input_config_data):
```

Comment:
>check_config_arch [](http://example.com/codeflow?start=4&length=17)

General naming observation: functions beginning with "check" make the reader assume we are verifying something and potentially returning/throwing an error or returning a boolean saying if the check succeeded or not. At least that would be my guess. In this case, we're trying to determine the architecture from the config, so I'd suggest something closer to `extract_config_arch`, `determine_config_arch`, etc.